third_party/USB-UIRT/Transmit/uu_irsage.cpp

```diff
@@ -0,0 +1,1008 @@
+    {
+      errStr = "Error: Unable to retrieve USB-UIRT Driver Version!";
+      unLoadDLL();
+      break;
+    }
+
+    if ((!fn_UUIRTGetDrvInfo(&drvVersion)) || (drvVersion != 0x0100))
+    {
+      errStr = "Error: USB-UIRT Driver Version (uuirtdrv.dll) is incompatible with this plugin. Please make sure you are using the latest version of plugin and driver!";
+      unLoadDLL();
+      break;
+    }
+
+    dllVersion = 0;
+    if (fnUUIRTOpenEx)
+      fn_UUIRTGetDrvVersion(&dllVersion);
+    if ((!fnUUIRTOpenEx) || (dllVersion < ((deviceNum > 0) ? (unsigned) 2670 : (unsigned) 2640)))
+    {
+      errStr = "Error: USB-UIRT Driver Version (uuirtdrv.dll) is incompatible with this plugin. Please make sure you are using version 2.6.4 or later of the USB-UIRT API driver!";
+      MessageBox(0, errStr, "Error", MB_OK);
+      unLoadDLL();
+      break;
+    }
+
+
+    if (deviceNum)
+      sprintf_s(devName, 256, "USB-UIRT-%d", deviceNum+1);
+    else
+      strcpy_s(devName, 256, "USB-UIRT");
+
+    hDrvHandle[deviceNum] = fnUUIRTOpenEx(devName, 0, NULL, NULL);
+    if (hDrvHandle[deviceNum] == INVALID_HANDLE_VALUE)
+    {
+      DWORD err;
+
+      err = GetLastError();
+
+      if (err == UUIRTDRV_ERR_NO_DLL)
+      {
+        errStr = "Error: Unable to find USB-UIRT Driver. Please make sure driver is Installed!";
+      }
+      else if (err == UUIRTDRV_ERR_NO_DEVICE)
+      {
+        errStr = "Error: Unable to connect to USB-UIRT device!  Please ensure device is connected to the computer!\n";
+      }
+      else if (err == UUIRTDRV_ERR_NO_RESP)
+      {
+        errStr = "Error: Unable to communicate with USB-UIRT device!  Please check connections and try again.  If you still have problems, try unplugging and reconnecting your USB-UIRT.  If problem persists, contact Technical Support!\n";
+      }
+      else if (err == UUIRTDRV_ERR_VERSION)
+      {
+        errStr = "Error: Your USB-UIRT's firmware is not compatible with the USB-UIRT driver installed on this machine. Please verify you are running the latest API DLL and that you're using the latest version of USB-UIRT firmware!  If problem persists, contact Technical Support!\n";
+      }
+      else
+      {
+        static char myStr[256];
+
+        sprintf_s(myStr,256, "Error: Unable to initialize USB-UIRT (unknown error) [%08lx]!\n",err);
+        DbgLog(myStr);
+        errStr = myStr;
+      }
+
+      unLoadDLL();
+
+      break;
+    }
+
+    break;
+  }
+
+  if (errStr)
+  {
+    return false;
+  }
+
+  return true;
+}
+
+
+TUNERSTUBDLL_API bool OpenDevice(int ComPort)
+{
+  FILE *fp = fopen("UU_LOG.ENABLE", "r"); // check if logging is enabled
+  if (fp != NULL)
+  {
+    fclose(fp);
+    logfp = fopen("uuirtlog.txt", "w");
+  }
+
+  return openHelper(0);
+}
+
+
+
+void closeHelper(int deviceNum)
+{
+  DbgLog("CloseDevice: dev #%d\n", deviceNum);
+
+  if ((hDrvHandle[deviceNum]) && (hDrvHandle[deviceNum] != INVALID_HANDLE_VALUE))
+    fnUUIRTClose(hDrvHandle[deviceNum]);
+  {
```

Comment:
Remove the braces here since I don't see any reason to change the scope for what executes below.

Reply:
yep